pkg/kocache/kocache_test.go

```diff
@@ -0,0 +1,252 @@
+package kocache
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStartOffline(t *testing.T) {
+	tests := map[string]struct {
+		initial *options
+		want    *options
+	}{
+		"Transition from offline should be offline": {
+			initial: &options{
+				StartOnline: false,
+			},
+			want: &options{
+				StartOnline: false,
+			},
+		},
+		"Transition from online should be offline": {
+			initial: &options{
+				StartOnline: true,
+			},
+			want: &options{
+				StartOnline: false,
+			},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			o := tt.initial
+			StartOffline()(o)
+			assert.Equal(t, tt.want.StartOnline, o.StartOnline)
+		})
+	}
+}
+
+func Test_applyDefaults(t *testing.T) {
+	tests := map[string]struct {
+		initial *options
+		want    *options
+	}{
+		"Default values should be as defined": {
+			initial: &options{},
+			want: &options{
+				ObjMemLimit:      1048576,
+				ObjHardLimit:     10485760,
+				CleanupThreshold: 10,
+				CleanupAge:       300000000000,
+				ErrorCleanUpAge:  30000000000,
+				CleanupInterval:  60000000000,
+				StartOnline:      true,
+				ModifyRequest:    nil,
+			},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			o := tt.initial
+			applyDefaults(o)
+			assert.Equal(t, tt.want, o)
+		})
+	}
+}
+
+func Test_koCache_GoOffline(t *testing.T) {
+	tests := map[string]struct {
+		startOnline bool
+	}{
+		"Starting online":  {startOnline: true},
+		"Starting offline": {startOnline: false},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			fn := func(o *options) {
+				if tt.startOnline {
+					o.StartOnline = true
+				}
+			}
+			ctx, cancel := context.WithCancelCause(context.Background())
+			defer cancel(errors.New("test ended"))
+			c := New(ctx, nil, "URL", fn)
+			c.GoOffline()
+			assert.False(t, c.IsOnline())
+		})
+	}
+}
+
+func Test_koCache_GoOnline(t *testing.T) {
+	tests := map[string]struct {
```

Comment:
why no `map[string]bool`?

Of even simpler: 
```go 
for _, b := range []bool{true, false} {
        t.Run(fmt.Sprintf("start offline: %s", strconv.FormatBool(v), func(t *testing.T) {
        // ...

Reply:
Done